lib/src/content/components/paginating_group/main_content_animated_builder.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+import 'package:wolt_modal_sheet/src/content/components/paginating_group/wolt_modal_sheet_page_transition_state.dart';
+
+class MainContentAnimatedBuilder extends StatefulWidget {
+  final AnimationController controller;
+  final GlobalKey outgoingOffstagedMainContentKey;
+  final GlobalKey incomingOffstagedMainContentKey;
+  final Widget child;
+  final bool forwardMove;
+  final double sheetWidth;
+  final WoltModalSheetPageTransitionState pageTransitionState;
+
+  const MainContentAnimatedBuilder({
+    required this.controller,
+    required this.child,
+    required this.outgoingOffstagedMainContentKey,
+    required this.incomingOffstagedMainContentKey,
+    required this.forwardMove,
+    required this.sheetWidth,
+    required this.pageTransitionState,
+    super.key,
+  });
+
+  @override
+  State<MainContentAnimatedBuilder> createState() => _MainContentAnimatedBuilderState();
+}
+
+class _MainContentAnimatedBuilderState extends State<MainContentAnimatedBuilder> {
+  Animation<double>? _sizeFactor;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.controller.addListener(() {
+      BuildContext? incomingContext = widget.incomingOffstagedMainContentKey.currentContext;
+      BuildContext? outgoingContext = widget.outgoingOffstagedMainContentKey.currentContext;
+      if (_sizeFactor == null &&
+          incomingContext?.mounted == true &&
+          outgoingContext?.mounted == true) {
+        _sizeFactor = widget.pageTransitionState.mainContentSizeFactor(
+          widget.controller,
+          incomingMainContentHeight: incomingContext!.size!.height,
+          outgoingMainContentHeight: outgoingContext!.size!.height,
+        );
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final screenWidth = MediaQuery.of(context).size.width;
+    final controller = widget.controller;
+    final pageTransitionState = widget.pageTransitionState;
+    return AnimatedBuilder(
+      animation: controller,
+      builder: (BuildContext context, Widget? _) {
```

Comment:
```suggestion
      builder: (BuildContext context, _) {
```

No type is needed when omitting